test/steps/data.go

```diff
@@ -180,12 +180,12 @@ func respinZookeeperWhenStuck(data *Data) {
 		case <-scanningPeriod.C:
 			namespaceExists, err := framework.IsNamespace(data.Namespace)
 			if err != nil {
-				return
+				framework.GetLogger(data.Namespace).Errorf("Error while checking existence of namespace for zookeeper workaround: %v", err)
 			}
 			if namespaceExists {
 				pods, err := framework.GetPodsWithLabels(data.Namespace, map[string]string{"app.kubernetes.io/name": "zookeeper"})
 				if err != nil {
-					return
+					framework.GetLogger(data.Namespace).Errorf("Error while getting zookeeper pods for zookeeper workaround: %v", err)
 				}
 
 				for _, pod := range pods.Items {
```

Comment:
if error here, `pods` is nil or ?
wouldn't that create an issue ?

Reply:
Good catch, fixed.
Now the iteration is executed in case err is nil.